packages/node_modules/pouchdb-abstract-mapreduce/src/index.js

```diff
@@ -285,7 +285,17 @@ function createAbstractMapReduce(localDocName, mapper, reducer, ddocValidator) {
         method: method,
         url: '_design/' + parts[0] + '/_view/' + parts[1] + params,
         body: body
-      }).then(postprocessAttachments(opts));
+      }).then(function (result) {
+        // fail the entire request if the result contains an error
+        var reduceError = result.rows.find(function (row) {
```

Comment:
I started with `some` but wanted the long-form reason to be included in the Error. Have pushed an update to make this clearer.